subsys/bluetooth/host/classic/shell/spp.c

```diff
@@ -0,0 +1,781 @@
+/*
+ * Copyright 2025 Xiaomi Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/settings/settings.h>
+
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/bluetooth/classic/rfcomm.h>
+#include <zephyr/bluetooth/classic/sdp.h>
+
+#include <zephyr/shell/shell.h>
+
+#include "host/shell/bt.h"
+#include "common/bt_shell_private.h"
+#include "common/bt_str.h"
+
+#define HELP_NONE "[none]"
+#define SPP_RFCOMM_MTU     CONFIG_BT_RFCOMM_L2CAP_MTU
+#define SDP_CLIENT_BUF_LEN 512
+
+enum __packed spp_state {
+	/* SPP_STATE_DISCONNECTED:
+	 * No active SPP connection. Resources are released or not allocated.
+	 */
+	SPP_STATE_DISCONNECTED,
+	/* SPP_STATE_CONNECTING:
+	 * Connection is in progress (e.g., SDP discovery or RFCOMM setup).
+	 * Not yet available for data transfer.
+	 */
+	SPP_STATE_CONNECTING,
+	/* SPP_STATE_CONNECTED:
+	 * RFCOMM DLC is established and SPP is ready for data transfer.
+	 */
+	SPP_STATE_CONNECTED,
+	/* SPP_STATE_DISCONNECTING:
+	 * Disconnection is in progress; cleanup and resource release pending.
+	 */
+	SPP_STATE_DISCONNECTING,
+};
+
+struct bt_spp;
+
+enum bt_spp_mode {
+	BT_SPP_MODE_UNKNOWN,
+	BT_SPP_MODE_UUID,
+	BT_SPP_MODE_CHANNEL,
+};
+
+/** @brief SPP profile application callback structure */
+struct bt_spp_ops {
```

Comment:
These look like definitions of a public API. Why are they in `shell/spp.c`?

Reply:
This is a historical code, removed